streams/src/test/java/org/apache/kafka/streams/integration/utils/EmbeddedKafkaCluster.java

```diff
@@ -118,8 +120,25 @@ private void putIfAbsent(final Properties props, final String propertyKey, final
      * Stop the Kafka cluster.
      */
     private void stop() {
+        if (brokers.length > 1) {
+            // delete the topics first to avoid cascading leader elections while shutting down the brokers
+            try {
+                final Set<String> topics = JavaConverters.setAsJavaSetConverter(brokers[0].kafkaServer().zkClient().getAllTopicsInCluster(false)).asJava();
+                final Admin adminClient = brokers[0].createAdminClient();
```

Comment:
We should use try-with-resource to make sure the AdminClient gets closed.

Reply:
Gah. Of course.